Dashboard/app/js/lib/controllers/survey-controllers.js

```diff
@@ -821,20 +821,10 @@ FLOW.questionGroupControl = Ember.ArrayController.create({
 
     questionGroup.deleteRecord();
 
-    // restore order of remaining groups
-    questionGroupsInSurvey = FLOW.store.filter(FLOW.QuestionGroup, function (item) {
-      return item.get('surveyId') == sId;
-    });
-
-    // restore order
-    questionGroupsInSurvey.forEach(function (item) {
-      if (item.get('order') > qgOrder) {
-        item.set('order', item.get('order') - 1);
-      }
-    });
+    // reorder the rest of the question groups
+    FLOW.questionControl.reorderQuestionGroups(sId, qgOrder, "up");
+    FLOW.questionControl.submitBulkQuestionGroupsReorder(sId);
```

Comment:
I suggest `reorderQuestionGroups` and `submitBulkQuestionGroupsReorder` belong to the `questionGroupControl`. It feels more coherent.